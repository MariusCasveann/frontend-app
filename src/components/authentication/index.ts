import LogoutButtonPresenter from './atoms/LogoutButton/LogoutButtonPresenter';
import UserMenu from './molecules/UserMenu/UserMenu';
import FakeAuth from './pages/FakeAuth/FakeAuth';

export { LogoutButtonPresenter, UserMenu, FakeAuth };
