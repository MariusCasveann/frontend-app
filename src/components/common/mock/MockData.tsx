import { GraphQLError } from 'graphql';
import { QueryAllProducts_allProducts } from './../../dashboard/atoms/ProductSelection/graphql/__generated__/QueryAllProducts';
import { QueryAllDomains_allDomains } from './../../dashboard/atoms/DomainSelection/graphql/__generated__/QueryAllDomains';
import { LightApplicationsOfProduct_lightApplicationsOfProduct } from './../../applications/graphql/__generated__/LightApplicationsOfProduct';
import { QueryBilling_billingQuery } from '../../dashboard/molecules/common/qraphql/__generated__/QueryBilling';
import {
    TeamByProductId_teamByProductId,
    TeamByProductId_teamByProductId_teamMembers
} from '../../teams/molecules/TeamMembers/graphql/__generated__/TeamByProductId';
import {
    QueryCurrentUser_currentUser,
    QueryCurrentUser_currentUser_roles
} from '../../authentication/molecules/UserProfile/graphql/__generated__/QueryCurrentUser';
import { SelectItemProps } from '../molecules/SearchableSelect/SearchableSelectPresenter';

export const mockGraphQLError: GraphQLError = {
    message: 'FATALITY',
    name: 'error',
    locations: undefined,
    path: undefined,
    nodes: undefined,
    positions: undefined,
    source: undefined,
    originalError: undefined,
    extensions: undefined
};

export const mockSelectedDomain: QueryAllDomains_allDomains = {
    id: 102,
    name: 'Engineering Platform',
    __typename: 'Domain'
};

export const mockSelectedProduct: QueryAllProducts_allProducts = {
    id: 50051,
    logo: null,
    name: 'a',
    __typename: 'LightProductDTO'
};

export const mockedApplications: LightApplicationsOfProduct_lightApplicationsOfProduct[] = [
    {
        id: 50051,
        staticPictureId: null,
        name: 'a',
        service: false,
        __typename: 'LightApplicationDTO'
    }
];

export const mockFunction = jest.fn();

export const mockDataEmpty: QueryBilling_billingQuery[] = [];

// tslint:disable-next-line:no-any
export const createMockedState = (loginAsProps?: Partial<{ data: any; loading: boolean; error: undefined }>): any => ({
    app: {
        authentication: {
            loginAs: {
                error: undefined,
                ...loginAsProps
            }
        },
        products: {
            selectedProduct: mockSelectedProduct
        }
    }
});

export const userProductMockedState = () => ({
    app: {
        authentication: {
            currentUser: mockCurrentUser
        },
        products: {
            selectedProduct: mockSelectedProduct
        }
    }
});

export const preSelectedProduct: QueryAllProducts_allProducts = {
    __typename: 'LightProductDTO',
    id: 'ALL',
    logo: 'ALL',
    name: 'All Products'
};

export const currentUserMockedState = () => ({
    app: {
        authentication: {
            currentUser: mockCurrentUser
        },
        products: {
            selectedProduct: preSelectedProduct
        }
    }
});

export const selectedProductMockedState = (
    loginAsProps?: Partial<{ data: any; loading: boolean; error: undefined }>
): any => ({
    app: {
        authentication: {
            loginAs: {
                error: undefined,
                ...loginAsProps
            }
        },
        products: {
            selectedProduct: mockSelectedProduct
        }
    }
});

export const buildDataExpectedEmptyResults = {
    labels: [],
    datasets: [
        {
            data: []
        }
    ]
};

export const mockErrorMessage = 'Error message';

export const mockTeamMember: TeamByProductId_teamByProductId_teamMembers = {
    __typename: 'User',
    email: 'test@mediamarktsaturn.com',
    firstName: 'Test',
    lastName: 'Test',
    id: '11',
    roles: [
        {
            name: 'WRITE_ALL',
            __typename: 'Role'
        },
        {
            name: 'a-admin',
            __typename: 'Role'
        }
    ]
};

export const mockTeamByProductId: TeamByProductId_teamByProductId = {
    __typename: 'Team',
    id: 2355,
    productOwner: {
        id: 11,
        firstName: 'TestUser',
        lastName: 'Test',
        __typename: 'User'
    },
    teamMembers: [
        {
            __typename: 'User',
            email: 'test1@mediamarktsaturn.com',
            firstName: 'Test',
            id: '1234',
            lastName: 'Test2',
            roles: [
                {
                    name: 'WRITE_ALL',
                    __typename: 'Role'
                },
                {
                    name: 'a-admin',
                    __typename: 'Role'
                }
            ]
        },
        {
            __typename: 'User',
            email: 'test2@mediamarktsaturn.com',
            firstName: 'Test2',
            id: '2234',
            lastName: 'Test2',
            roles: [
                {
                    name: 'WRITE_ALL',
                    __typename: 'Role'
                },
                {
                    name: 'a-admin',
                    __typename: 'Role'
                },
                {
                    name: 'test-admin',
                    __typename: 'Role'
                }
            ]
        }
    ]
};

export const mockCurrentUser: QueryCurrentUser_currentUser = {
    __typename: 'UserDTO',
    activated: true,
    createdBy: 'anonymousUser',
    createdDate: '2019-05-27T10:53:29.944312Z',
    email: 'test@mediamarktsaturn.com',
    firstName: 'Test',
    id: 49050,
    imageContent: null,
    langKey: null,
    lastModifiedBy: 'Test',
    lastModifiedDate: '2019-06-28T05:56:51.175813Z',
    lastName: 'Test',
    login: 'Test',
    roles: [
        {
            concretePermissions: [],
            id: 1300,
            name: 'DP God',
            readOnly: true,
            __typename: 'Role'
        },
        {
            concretePermissions: [],
            id: 4651,
            name: 'READ_ALL',
            readOnly: false,
            __typename: 'Role'
        },
        {
            concretePermissions: [],
            id: 69002,
            name: 'a-read',
            readOnly: true,
            __typename: 'Role'
        }
    ],
    staticPictureId: null,
    subscribedToNewsletter: true
};

export const mockEmptyTeam: TeamByProductId_teamByProductId = {
    __typename: 'Team',
    id: null,
    productOwner: null,
    teamMembers: []
};

export const mockCurrenUserRole: QueryCurrentUser_currentUser_roles = {
    concretePermissions: [],
    id: 1300,
    name: 'DP God',
    readOnly: true,
    __typename: 'Role'
};

export const mockForm = {
    getFieldDecorator: () => () => false,
    getFieldError: mockFunction,
    getFieldsError: mockFunction,
    getFieldsValue: mockFunction,
    getFieldValue: mockFunction,
    isFieldValidating: mockFunction,
    setFields: mockFunction,
    setFieldsValue: mockFunction,
    validateFields: mockFunction,
    validateFieldsAndScroll: mockFunction,
    isFieldsTouched: mockFunction,
    isFieldTouched: mockFunction,
    resetFields: mockFunction
};

export const mockHistory = {
    length: 7,
    push: mockFunction,
    action: 'PUSH',
    location: {
        hash: '',
        pathname: '/path',
        search: '',
        state: {}
    },
    replace: mockFunction,
    go: mockFunction,
    goBack: mockFunction,
    goForward: mockFunction,
    block: mockFunction,
    createHref: mockFunction,
    listen: mockFunction
};

export const mockSelectData: SelectItemProps[] = [
    {
        id: '351952',
        label: 'Test test'
    },
    {
        id: '80503',
        label: 'Test1 test1'
    },
    {
        id: '412605',
        label: 'Test2 test2'
    }
];
