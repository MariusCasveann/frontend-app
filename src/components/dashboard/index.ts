import DashboardPresenter from './pages/GovernanceDashboard/GovernanceDashboardPresenter';

export { DashboardPresenter };
