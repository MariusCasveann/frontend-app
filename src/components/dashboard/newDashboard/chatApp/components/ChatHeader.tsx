import * as React from 'react';
import { useState } from 'react';
import { Icon } from 'antd';
import moment from 'moment';
import { UserType } from '../mockDataForChat';

const headerStyle = {
    backgroundColor: 'grey',
    display: 'flex',
    justifyContent: 'space-between',
    padding: '5px 5px 0 5px'
};

interface HeaderProps {
    currentUser: UserType;
}

export const ChatHeader = (props: HeaderProps) => {
    const { firstname, lastname } = props.currentUser;
    const [currentTime, setCurrentTime] = useState<string>('');

    setInterval(() => {
        return setCurrentTime(`${moment().format('LTS')}`);
    }, 1000);

    return (
        <div style={headerStyle}>
            <h2>CHAT PLATFORM</h2>
            <div style={{ display: 'flex', flexDirection: 'column' }}>
                <h3 style={{ color: 'white', marginRight: 30, marginBottom: 0 }}>
                    <Icon type="user" />{' '}
                    <b>
                        {firstname} {lastname}
                    </b>
                </h3>
                <p style={{ margin: 0, color: 'white' }}>Time: {currentTime}</p>
            </div>
        </div>
    );
};
