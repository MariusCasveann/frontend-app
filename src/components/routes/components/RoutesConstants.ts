// components
import UserProfile from '../../authentication/molecules/UserProfile/UserProfile';
import { ComponentClass, FunctionComponent } from 'react';
import PageNotFound from '../atoms/PageNotFound';
import { NewDashboard } from '../../dashboard/newDashboard/NewDashboard';
import { EditUserPage } from '../../dashboard/newDashboard/components/EditUserPage';
import { ChatApplication } from '../../dashboard/newDashboard/chatApp/components/ChatApplication';

// constants
import {
    ADMIN_EXTERNAL_TOOLS_ROUTE,
    BUSINESS_DOMAIN_ROUTE,
    DEPLOYMENT_STAGE_ROUTE,
    GOOGLE_FOLDER_ROUTE,
    NETWORK_READ,
    SERVICE_SECTION_ROUTE
} from './adminRoutes.constants';
import DashboardTabsPresenter from '../../dashboard/pages/DashboardTabs/DashboardTabsPresenter';
import LightApplications from '../../applications/LightApplications';
import CreateTeam from '../../createTeam/CreateTeam';
import Team from '../../teams/molecules/TeamMembers/Team';

const DP_GOD = 'DP God';

export const ALWAYS = 'ALWAYS';

const READ = 'READ';

export const WRITE = 'WRITE';

export const SERVICE_SECTION = 'SERVICE_SECTION';

const CLOUD_FOUNDATION_PLATFORM = 'cloud-foundation-platform';

export const adminRoles = [DP_GOD];

const ALLOWED_ROUTES_FOR_PRODUCT: string[] = [...adminRoles, READ, WRITE];

// paths
const DASHBOARD_PATH = '/dashboard';
const USER_EDIT_PATH = '/user-edit';
const APPLICATIONS_PATH = '/applications';
const BILLING_PATH = '/billing';
const APPLICATION_EDIT_PATH = '/edit-application/:id';
const PRODUCT_EDIT_PATH = '/edit-product/:id';
const BUILD_DETAILS_PATH = '/build-details';
const BUILDS_PATH = '/application/:id';
const LOGIN_PATH = '/login';
const NEW_APPLICATION_PATH = '/new-application';
const REGISTER_PATH = '/register';
const RESET_PASSWORD_PATH = '/reset-password/:id';
const ROLES_PATH = '/roles';
const ROLE_PATH = '/role/:id';
const SERVICE_CATALOGUE_PATH = '/service-catalogue';
const DOMAIN_PATH = '/domain';
const SERVICE_DETAILS_PATH = '/service-details/:id';
const NEW_PRODUCT_PATH = '/new-product';
const PRIVACY_PATH = '/privacy';
const TEAM_PATH = '/team';
const CREATE_TEAM_PATH = '/create-team';
const USERS_PATH = '/users';
const PRODUCT_DETAILS_PATH = '/product-details';
const SERVICE_SECTION_PATH = '/service-section';
const EXTERNAL_TOOLS_PATH = '/external-tools';
const NEW_DASHBOARD_PATH = '/new-dashboard';
const EDIT_USER_PATH = '/new-dashboard/edit-user/:name'; // :title  - marchez o valoare dinamica
const CHAT_APP = '/chat-app';

export interface RouteI {
    allowedRoles: string[];
    title?: string;
    path: string;
    name?: string;
    component?: ComponentClass | FunctionComponent;
    checkFoundation?: boolean;
    checkServiceSectionPermission?: boolean;
    checkAdminRole?: boolean;
}

export interface Navigation {
    icon?: string;
    label?: string;
    children?: Navigation[];
    navLink: string;
    checkProductWasSelected?: boolean;
}

const adminRoutes: RouteI[] = [
    {
        allowedRoles: [...adminRoles, NETWORK_READ],
        checkAdminRole: true,
        component: PageNotFound,
        path: DEPLOYMENT_STAGE_ROUTE
    },
    {
        allowedRoles: adminRoles,
        checkAdminRole: true,
        component: PageNotFound,
        path: ADMIN_EXTERNAL_TOOLS_ROUTE
    },
    {
        allowedRoles: adminRoles,
        checkAdminRole: true,
        component: PageNotFound,
        path: GOOGLE_FOLDER_ROUTE
    },
    {
        allowedRoles: adminRoles,
        checkAdminRole: true,
        component: PageNotFound,
        path: SERVICE_SECTION_ROUTE
    },
    {
        allowedRoles: adminRoles,
        checkAdminRole: true,
        component: PageNotFound,
        path: BUSINESS_DOMAIN_ROUTE
    }
];

export const routes: RouteI[] = [
    ...adminRoutes,
    {
        allowedRoles: [DP_GOD],
        checkAdminRole: true,
        component: DashboardTabsPresenter,
        path: DASHBOARD_PATH
    },
    {
        allowedRoles: [ALWAYS],
        checkAdminRole: true,
        component: UserProfile,
        path: USER_EDIT_PATH
    },
    {
        allowedRoles: ALLOWED_ROUTES_FOR_PRODUCT,
        title: 'Applications',
        path: APPLICATIONS_PATH,
        checkFoundation: true,
        component: LightApplications
    },
    {
        allowedRoles: [...ALLOWED_ROUTES_FOR_PRODUCT, CLOUD_FOUNDATION_PLATFORM],
        title: 'Billing',
        path: BILLING_PATH,
        checkFoundation: true,
        component: PageNotFound
    },
    {
        allowedRoles: ALLOWED_ROUTES_FOR_PRODUCT,
        title: 'Edit Application',
        path: APPLICATION_EDIT_PATH,
        checkFoundation: true,
        component: PageNotFound
    },
    {
        allowedRoles: ALLOWED_ROUTES_FOR_PRODUCT,
        title: 'Edit Product',
        path: PRODUCT_EDIT_PATH,
        component: PageNotFound
    },
    {
        allowedRoles: ALLOWED_ROUTES_FOR_PRODUCT,
        title: 'Build-Details',
        path: BUILD_DETAILS_PATH,
        checkFoundation: true,
        component: PageNotFound
    },
    {
        allowedRoles: ALLOWED_ROUTES_FOR_PRODUCT,
        title: 'Builds',
        path: BUILDS_PATH,
        checkFoundation: true,
        component: PageNotFound
    },
    {
        allowedRoles: [ALWAYS],
        title: 'Login',
        path: LOGIN_PATH,
        component: PageNotFound
    },
    {
        allowedRoles: ALLOWED_ROUTES_FOR_PRODUCT,
        title: 'New Application',
        path: NEW_APPLICATION_PATH,
        checkFoundation: true,
        component: PageNotFound
    },
    {
        allowedRoles: [ALWAYS],
        title: 'Register',
        path: REGISTER_PATH,
        component: PageNotFound
    },
    {
        allowedRoles: [ALWAYS],
        title: 'Reset Password',
        path: RESET_PASSWORD_PATH,
        component: PageNotFound
    },
    {
        allowedRoles: adminRoles,
        component: PageNotFound,
        checkAdminRole: true,
        title: 'Roles',
        path: ROLES_PATH
    },
    {
        allowedRoles: adminRoles,
        component: PageNotFound,
        checkAdminRole: true,
        title: 'Role',
        path: ROLE_PATH
    },
    {
        allowedRoles: [ALWAYS],
        title: 'Service Catalogue',
        path: SERVICE_CATALOGUE_PATH,
        component: PageNotFound
    },
    {
        allowedRoles: [ALWAYS],
        title: 'Dashboard',
        path: NEW_DASHBOARD_PATH,
        component: NewDashboard
    },
    {
        allowedRoles: [ALWAYS],
        title: 'Chat application',
        path: CHAT_APP,
        component: ChatApplication
    },
    {
        allowedRoles: [ALWAYS],
        title: 'Edit user',
        path: EDIT_USER_PATH,
        component: EditUserPage
    },
    {
        allowedRoles: ALLOWED_ROUTES_FOR_PRODUCT,
        title: 'PageNotFound',
        path: DOMAIN_PATH,
        component: PageNotFound
    },
    {
        allowedRoles: [ALWAYS],
        title: 'Service Details',
        path: SERVICE_DETAILS_PATH,
        component: PageNotFound
    },
    {
        allowedRoles: ALLOWED_ROUTES_FOR_PRODUCT,
        title: 'New Product',
        path: NEW_PRODUCT_PATH,
        component: PageNotFound
    },
    {
        allowedRoles: [ALWAYS],
        title: 'Privacy Policy',
        path: PRIVACY_PATH,
        component: PageNotFound
    },
    {
        allowedRoles: ALLOWED_ROUTES_FOR_PRODUCT,
        title: 'Team',
        path: TEAM_PATH,
        component: Team
    },
    {
        allowedRoles: ALLOWED_ROUTES_FOR_PRODUCT,
        title: 'Create Team',
        path: CREATE_TEAM_PATH,
        component: CreateTeam
    },
    {
        allowedRoles: adminRoles,
        checkAdminRole: true,
        component: PageNotFound,
        path: USERS_PATH,
        title: 'Users'
    },
    {
        allowedRoles: [ALWAYS],
        title: 'Edit profile',
        path: USER_EDIT_PATH,
        component: PageNotFound
    },
    {
        allowedRoles: ALLOWED_ROUTES_FOR_PRODUCT,
        title: 'Product Details',
        path: PRODUCT_DETAILS_PATH,
        component: PageNotFound
    },
    {
        allowedRoles: [...adminRoles, WRITE],
        checkServiceSectionPermission: true,
        title: 'Service Section',
        path: SERVICE_SECTION_PATH,
        component: PageNotFound
    },
    {
        allowedRoles: [...adminRoles, WRITE],
        title: 'External Tools',
        path: EXTERNAL_TOOLS_PATH,
        component: PageNotFound
    }
];

export const navigationOptions: Navigation[] = [
    {
        icon: 'appstore',
        label: 'Dashboard',
        navLink: DASHBOARD_PATH
    },
    {
        icon: 'database',
        label: 'Applications',
        navLink: APPLICATIONS_PATH,
        checkProductWasSelected: true
    },
    {
        icon: 'credit-card',
        label: 'Billing',
        navLink: BILLING_PATH,
        checkProductWasSelected: true
    },
    {
        icon: 'control',
        label: 'Roles',
        navLink: ROLES_PATH
    },
    {
        icon: 'team',
        label: 'Team',
        navLink: TEAM_PATH,
        checkProductWasSelected: true
    },
    {
        icon: 'settings',
        label: 'External Tools',
        navLink: EXTERNAL_TOOLS_PATH,
        checkProductWasSelected: true
    },
    {
        icon: 'search',
        label: 'Product Details',
        navLink: PRODUCT_DETAILS_PATH,
        checkProductWasSelected: true
    },
    {
        icon: 'appstore',
        label: 'Service Catalogue',
        navLink: SERVICE_CATALOGUE_PATH
    },
    {
        icon: 'laptop',
        label: 'New Dashboard',
        navLink: NEW_DASHBOARD_PATH
    },
    {
        icon: 'message',
        label: 'Chat',
        navLink: CHAT_APP
    },
    {
        icon: 'appstore',
        label: 'Business Domain',
        navLink: DOMAIN_PATH
    },
    {
        icon: 'appstore',
        label: 'Service section',
        navLink: SERVICE_SECTION_PATH
    },
    {
        icon: 'info',
        label: 'Super Admin Page',
        navLink: '/admin',
        children: [
            {
                icon: 'bundle',
                label: 'Deployment stages',
                navLink: DEPLOYMENT_STAGE_ROUTE
            },
            {
                icon: 'bundle',
                label: 'External tools',
                navLink: ADMIN_EXTERNAL_TOOLS_ROUTE
            },
            {
                icon: 'bundle',
                label: 'Google Folders',
                navLink: GOOGLE_FOLDER_ROUTE
            },
            {
                icon: 'bundle',
                label: 'Business Domain',
                navLink: BUSINESS_DOMAIN_ROUTE
            }
        ]
    }
];
