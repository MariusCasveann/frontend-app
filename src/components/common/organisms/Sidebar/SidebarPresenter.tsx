import React from 'react';
import { debounce } from 'lodash';
import { Menu, Layout, Icon } from 'antd';
import { createHashHistory } from 'history';

// constants
import { Navigation, navigationOptions, RouteI } from '../../../routes/components/RoutesConstants';

// components
import SelectedProduct from './SelectedProduct';

// css
import './Sidebar.css';

// interfaces
import { shouldRenderNavItem } from '../../../routes/helpers/routesHelper';
import { QueryAllProducts_allProducts } from '../../../dashboard/atoms/ProductSelection/graphql/__generated__/QueryAllProducts';

const { Sider } = Layout;
const history = createHashHistory();

const renderMenuItem = (item: Navigation) => (
    <Menu.Item key={item.navLink} onClick={() => history.push(item.navLink)}>
        <Icon type={item.icon} />
        {item.label}
    </Menu.Item>
);

interface SidebarPresenterProps {
    allowedRoutes: RouteI[];
    selectedProduct: QueryAllProducts_allProducts;
}

export default class SidebarPresenter extends React.Component<SidebarPresenterProps> {
    public state = {
        collapse: false
    };

    public componentDidMount() {
        window.addEventListener('resize', this.onResize);
    }

    private setCollapse = () => {
        this.setState({ collapse: !this.state.collapse });
    };

    private setCollapseOnWindowResize = () => {
        if (document.documentElement.clientWidth < 768 && !this.state.collapse) {
            this.setState({
                collapse: true
            });
        } else if (document.documentElement.clientWidth > 768 && this.state.collapse) {
            this.setState({ collapse: false });
        }
    };

    private onResize = debounce(this.setCollapseOnWindowResize, 100);

    public render() {
        const { collapse } = this.state;
        const { allowedRoutes, selectedProduct } = this.props;

        return (
            <Sider collapsible={true} collapsed={collapse} onCollapse={() => this.setCollapse()}>
                <Menu mode="inline" theme="dark" style={{ height: '100%' }}>
                    <SelectedProduct showSelect={!collapse} />
                    {navigationOptions.map((item: Navigation) => {
                        if (shouldRenderNavItem(item, allowedRoutes, selectedProduct)) {
                            if (!item.children) {
                                return renderMenuItem(item);
                            }

                            return (
                                <Menu.SubMenu
                                    key={item.navLink}
                                    className="admin-submenu"
                                    title={
                                        <React.Fragment>
                                            <Icon type={item.icon} />
                                            <span>{item.label}</span>
                                        </React.Fragment>
                                    }
                                >
                                    {item.children.map((child: Navigation) => renderMenuItem(child))}
                                </Menu.SubMenu>
                            );
                        }

                        return false;
                    })}
                </Menu>
            </Sider>
        );
    }
}
