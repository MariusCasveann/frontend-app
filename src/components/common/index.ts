import Header from './../common/molecules/Header/HeaderPresenter';
import SidebarPresenter from './organisms/Sidebar/SidebarPresenter';
import Breadcrumb from './molecules/Breadcrumb/BreadcrumbPresenter';

export { Breadcrumb, Header, SidebarPresenter };
