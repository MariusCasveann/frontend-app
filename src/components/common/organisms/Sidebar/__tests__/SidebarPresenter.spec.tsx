import ShallowRenderer from 'react-test-renderer/shallow';
import * as React from 'react';
import { shallow } from 'enzyme';
import { RouteI } from '../../../../routes/components/RoutesConstants';
import SidebarPresenter from '../SidebarPresenter';
import Sider from 'antd/lib/layout/Sider';
import { mockSelectedProduct } from '../../../mock/MockData';

const mockAllowedRoutes: RouteI[] = [
    {
        allowedRoles: ['DP God', 'NETWORK_READ'],
        checkAdminRole: true,
        path: '/admin/deployment-stages'
    },
    {
        allowedRoles: ['DP God'],
        checkAdminRole: true,
        path: '/admin/external-tools'
    },
    {
        allowedRoles: ['DP God', 'READ', 'WRITE'],
        checkAdminRole: true,
        path: '/application/:id'
    }
];

// tslint:disable-next-line:no-any
const _ShallowRenderer: any = ShallowRenderer;
const renderer = new _ShallowRenderer();

const tree = renderer.render(
    <SidebarPresenter selectedProduct={mockSelectedProduct} allowedRoutes={mockAllowedRoutes} />
);
const result = renderer.getRenderOutput();
const wrapper = shallow(<SidebarPresenter selectedProduct={mockSelectedProduct} allowedRoutes={mockAllowedRoutes} />);

describe('SidebarPresenter', () => {
    it('renders correctly', () => {
        expect(tree).toMatchSnapshot();
    });
    it('renders a Sider', () => {
        expect(result.type).toBe(Sider);
    });
    it('renders a Menu', () => {
        expect(wrapper.find('Menu').exists()).toBe(true);
    });
    it('renders the number of MenuItems provided', () => {
        expect(wrapper.find('MenuItem').length).toEqual(4);
    });
    it('renders the number of Icons provided', () => {
        expect(wrapper.find('Icon').length).toEqual(4);
    });
    it('renders SubMenu', () => {
        expect(wrapper.find('SubMenu').exists()).toBe(true);
    });
});
