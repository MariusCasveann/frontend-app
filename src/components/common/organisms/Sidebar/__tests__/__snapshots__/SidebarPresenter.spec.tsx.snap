// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarPresenter renders correctly 1`] = `
<Sider
  collapsed={false}
  collapsible={true}
  onCollapse={[Function]}
>
  <Menu
    mode="inline"
    style={
      Object {
        "height": "100%",
      }
    }
    theme="dark"
  >
    <Apollo(Connect(_default))
      showSelect={true}
    />
    <SubMenu
      className="admin-submenu"
      title={
        <React.Fragment>
          <Icon
            type="info"
          />
          <span>
            Super Admin Page
          </span>
        </React.Fragment>
      }
    >
      <MenuItem
        onClick={[Function]}
      >
        <Icon
          type="bundle"
        />
        Deployment stages
      </MenuItem>
      <MenuItem
        onClick={[Function]}
      >
        <Icon
          type="bundle"
        />
        External tools
      </MenuItem>
      <MenuItem
        onClick={[Function]}
      >
        <Icon
          type="bundle"
        />
        Google Folders
      </MenuItem>
      <MenuItem
        onClick={[Function]}
      >
        <Icon
          type="bundle"
        />
        Business Domain
      </MenuItem>
    </SubMenu>
  </Menu>
</Sider>
`;
